### PR TITLE
Legend addition with component separation and changes in Readme

### PR DESCRIPTION
**Pull request by aarathyKeyvalue**

### CODE REVIEW

File: src/lib/dot-matrix/Chart.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      style={{
+        width: `${getContainerWidth(columns)}rem`,
+        ...getStyles(Elements.DotsContainer)
+      }}
+    >
+      {data?.map((eachPoint: DataPointType, rowIndex: number) => (
+        <React.Fragment key={v4()}>
+          {eachPoint && Array.apply(null, Array(getNumberOfDots(eachPoint, rows, columns, total))).map((item: null, columnIndex: number) => (
+            <div id="dot-matrix-dots" key={v4()}>
+              {(columnIndex === 0 && rowIndex > 0 && overlappingValues[rowIndex - 1] < 1 && overlappingValues[rowIndex - 1] !== 0 && (
```
Comment (arunghosh-kv) [inline src/lib/dot-matrix/Chart.tsx]: Can we move this logic an `hasOverlapping(overlappingValues, rowIndex, columnIndex)`?

File: src/lib/dot-matrix/Chart.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  } = dimensions;
+  return (
+    <div
+      className={classes.dotsContainer}
+      style={{
+        width: `${getContainerWidth(columns)}rem`,
+        ...getStyles(Elements.DotsContainer)
+      }}
+    >
+      {data?.map((eachPoint: DataPointType, rowIndex: number) => (
```
Comment (arunghosh-kv) [inline src/lib/dot-matrix/Chart.tsx]: Instead of eachPoint, may be dataItem?

File: src/lib/dot-matrix/DotMatrix.tsx [hunk 3]
```diff
@@ -34,42 +32,52 @@ const DotMatrix = (props: DotMatrixPropType): JSX.Element => {
   };
 
 
+  const getLegendPosition = (): {flexDirection: string, alignItems: string} => {
+    let flexDirection = '';
+    let alignItems = 'end';
+    switch (legendPosition) {
```
Comment (arunghosh-kv) [inline src/lib/dot-matrix/DotMatrix.tsx]: Replace with Object?

File: src/lib/dot-matrix/DotMatrix.tsx [hunk 4]
```diff
@@ -34,42 +32,52 @@ const DotMatrix = (props: DotMatrixPropType): JSX.Element => {
   };
 
 
+  const getLegendPosition = (): {flexDirection: string, alignItems: string} => {
```
Comment (arunghosh-kv) [inline src/lib/dot-matrix/DotMatrix.tsx]: Do we need it inside the component?

File: src/lib/dot-matrix/DotMatrix.tsx [hunk 5]
```diff
@@ -34,42 +32,52 @@ const DotMatrix = (props: DotMatrixPropType): JSX.Element => {
...
-                    key={v4()}
-                    id={`each-category-${rowIndex}-${columnIndex}`}
-                  />
-                )}
-              </div>
-            ))}
-          </React.Fragment>
-        ))}
+        <Chart
+          getStyles={getStyles}
```
Comment (arunghosh-kv) [inline src/lib/dot-matrix/DotMatrix.tsx]: pass styles and define getStyles inside Chart

File: src/lib/dot-matrix/DotMatrix.tsx [hunk 6]
```diff
@@ -1,29 +1,27 @@
...
+    showLegend,
+    legendPosition = DEFAULT_LEGEND_POSITION
   } = props;
 
-  const {
-    rows = DEFAULT_ROWS,
-    columns = DEFAULT_ROWS
-  } = dimensions;
-
   const [data, total, overlappingValues] = useDotMatrix(dataPoints, dimensions);
```
Comment (arunghosh-kv) [inline src/lib/dot-matrix/DotMatrix.tsx:26]: Possible to move this inside Chart?

Comment (aarathyKeyvalue) [reply src/lib/dot-matrix/DotMatrix.tsx:26]: We need data inside legend too

File: src/lib/dot-matrix/Chart.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    data,
+    overlappingValues,
+    total
+  } = props;
+  const {
+    rows = DEFAULT_ROWS,
+    columns = DEFAULT_COLUMNS
+  } = dimensions;
+
+  const hasOverlapping = (values: number[], indexRow: number, indexColumn: number): boolean => (
```
Comment (arunghosh-kv) [inline src/lib/dot-matrix/Chart.tsx]: We can move this out of the component? Have no state dependency

**Review by arunghosh-kv — APPROVED** [review]

**Pull request merged**